pkg/prometheus/agent/operator.go

```diff
@@ -856,7 +856,7 @@ func (c *Operator) syncStatefulSet(ctx context.Context, key string, p *monitorin
 	return nil
 }
 
-func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *monitoringv1alpha1.PrometheusAgent, cg *prompkg.ConfigGenerator, store *assets.StoreBuilder) error {
+func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *monitoringv1alpha1.PrometheusAgent, cg *prompkg.ConfigGenerator, store *assets.StoreBuilder, daemonSet bool) error {
```

Comment:
(nit) I think an enum would be more idiomatic

like 

```
const (
	DaemonSet   WorkloadType = "DaemonSet"
	StatefulSet WorkloadType = "StatefulSet"
)

switch workload {
	case DaemonSet:
		fmt.Println("The workload is a DaemonSet")
	case StatefulSet:
		fmt.Println("The workload is a StatefulSet")
	default:
		fmt.Println("Unknown workload type")
	}
```

Just an example

Reply:
Yeah we repeat these everywhere, not only in this PR. I'll take a look later how far we can generalize^^

Reply:
I'm not sure that we need an extra argument since the information is already present in the PrometheusAgent object.

Reply:
Aha! Funny silly mistake...